### PR TITLE
Split up integration tests with structure; Add watcher

### LINKED ISSUE #18: Use Watchman to auto-run tests when source changes
IDEs sometimes have this feature, but [Watchman](https://github.com/facebook/watchman) could add this for everyone using a text editor and CLI.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/5878547-use-watchman-to-auto-run-tests-when-source-changes?utm_campaign=plugin&utm_content=tracker%2F1487757&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F1487757&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #81: Research the arguments for/against tests in the package
Currently, the template suggests putting the tests in the package, but I'd like to remember the main arguments.

The first I can think of is that if you have doctests you can simply run your entire package through your test runner.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/13926213-research-the-arguments-for-against-tests-in-the-package?utm_campaign=plugin&utm_content=tracker%2F1487757&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F1487757&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by jacebrowning**
Fixes #18
Fixes #81 

Thanks @Ross-Hunter for showing me that this was a good idea.

### CODE REVIEW

**Comment by jacebrowning** [comment]
@joshfriend Can you confirm that I did this correctly by cookiecutting an example using both `nose` and `pytest` and running `$ make watch` in each?

File: {{cookiecutter.project_name}}/scent.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        (('make', 'test-unit'), "Unit Tests"),
+        (('make', 'test-int'), "Integration Tests"),
+        (('make', 'test-all'), "Combined Tests"),
+    ), start=1):
+
+        failure = subprocess.call(command)
+
+        if failure:
+            mark = "❌" * count
+            Notifier.notify(mark + " [FAIL] " + mark, title=title)
```
Comment (joshfriend) [inline {{cookiecutter.project_name}}/scent.py]: add `group=1234` or some integer so each loop iteration doesn't create a separate notification and clutter up notification center. If the group is unique per test run, you will only get one notification per run. Using the epoch time is a good way to do this:

```patch
diff --git a/{{cookiecutter.project_name}}/scent.py b/{{cookiecutter.project_name}}/scent.py
index 6d7a84c..407acda 100644
--- a/{{cookiecutter.project_name}}/scent.py
+++ b/{{cookiecutter.project_name}}/scent.py
@@ -3,12 +3,13 @@
 {% endif -%}
 import os
 import subprocess
+import time

 from sniffer.api import select_runnable, file_validator, runnable
 from pync import Notifier


-watch_paths = ['{{cookiecutter.package_name}}/', 'tests/']
+watch_paths = ['demo/', 'tests/']


 @select_runnable('python_tests')
@@ -20,7 +21,7 @@ def py_files(filename):

 @runnable
 def python_tests(*args):
-
+    gid = int(time.time())
     for count, (command, title) in enumerate((
         (('make', 'test-unit'), "Unit Tests"),
         (('make', 'test-int'), "Integration Tests"),
@@ -31,10 +32,10 @@ def python_tests(*args):

         if failure:
             mark = "❌" * count
-            Notifier.notify(mark + " [FAIL] " + mark, title=title)
+            Notifier.notify(mark + " [FAIL] " + mark, title=title, group=gid)
             return False
         else:
             mark = "✅" * count
-            Notifier.notify(mark + " [PASS] " + mark, title=title)
+            Notifier.notify(mark + " [PASS] " + mark, title=title, group=gid)

     return True
```

File: {{cookiecutter.project_name}}/scent.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            return False
+        else:
+            mark = "✅" * count
```
Comment (joshfriend) [inline {{cookiecutter.project_name}}/scent.py]: If you use the escaped encoding of these glyphs (`\xe2\x9d\x8c` and `\xe2\x9c\x85`, respectively) this file wont show up funny in some editors and you probably also wont need the conditional encoding declaration.

**Comment by joshfriend** [comment]
works fine for me

File: {{cookiecutter.project_name}}/scent.py (hunk 2 above)
Comment (jacebrowning) [inline {{cookiecutter.project_name}}/scent.py]: Do you have an example of an editor that can't handle this? Personally, I think it's kind of neat to see the actual unicode character.

Comment (joshfriend) [inline {{cookiecutter.project_name}}/scent.py]: It's probably a font issue most of the time, but terminal editors (`vi`, `nano`, etc) do this for me:


Also doesn't play nice with some color schemes (sublime here):

**Comment by jacebrowning** [comment]
@joshfriend I think this will handle the non-Mac case now.

File: {{cookiecutter.project_name}}/scent.py (hunk 2 above)
Comment (jacebrowning) [inline {{cookiecutter.project_name}}/scent.py]: Interesting. But considering this file is going to be edited very infrequently, I'm going to leave this for now.

**Pull request merged**